testhelpers/server_factory.go

```diff
@@ -0,0 +1,45 @@
+package testhelpers
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type HTTPTestResponse struct {
+	Code int
+	Body string
+}
+
+type HTTPTestRequest struct {
+	Path   string
+	Method string
+	Body   string
+}
+
+func BuildHTTPTestServer(request *HTTPTestRequest, response *HTTPTestResponse, requestLabel TestRequestLabel) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer GinkgoRecover()
+
+		TestRequestTracker <- requestLabel
+
+		if request.Path != "" {
+			Expect(r.URL.Path).To(Equal(request.Path))
+		}
+		if request.Method != "" {
+			Expect(r.Method).To(Equal(request.Method))
+		}
+
+		body, err := ReadHTTPBody(r.Body)
+		Expect(err).To(BeNil())
+		if request.Body != "" {
+			Expect(body).To(MatchJSON(request.Body))
+		}
+
+		w.WriteHeader(response.Code)
+		fmt.Fprintln(w, response.Body)
+	}))
```

Comment:
yeah, that's true, i too wasn't happy with that coupling, but thought of taking small steps to restructuring this. turns out, using `ghttp` package was an excellent suggestion, didn't know about that. :+1: